topo/node/juniper/juniper.go

```diff
@@ -109,6 +113,38 @@ func (n *Node) GRPCConfig() []string {
 	}
 }
 
+// Waits and retries until CLI config mode is up and config is applied
+func (n *Node) waitConfigInfraReadyAndPushConfigs(configs []string) error {
```

Comment:
Can you add a unit test that validates this new behavior?

Other than that this PR looks good

Reply:
Done